File: app/admin/router/system/standardPhrase.go
package system

import (
	"github.com/gin-gonic/gin"
	api "haocean/health-enforcement/app/admin/api/system"
	"haocean/health-enforcement/app/core/utils/jwt"
)

// InitStandardPhrase 初始化规范用语路由
func InitStandardPhrase(e *gin.Engine) {
	v := e.Group("system")
	{
		// 公开路由（可缓存）
		v.GET("/standard-phrase/supervision-type/list", api.ListSupervisionType)
		v.GET("/standard-phrase/supervision-type/:id", api.GetSupervisionType)

		v.GET("/standard-phrase/category/list", api.ListCategory)
		v.GET("/standard-phrase/category/:id", api.GetCategory)

		v.GET("/standard-phrase/item/list", api.ListItem)
		v.GET("/standard-phrase/item/:id", api.GetItem)

		v.GET("/standard-phrase/content/list", api.ListContent)
		v.GET("/standard-phrase/content/:id", api.GetContent)

		// 搜索和树形结构
		v.GET("/standard-phrase/search", api.SearchStandardPhrase)
		v.GET("/standard-phrase/tree", api.GetFullTree)

		// 需要认证的路由
		auth := v.Group("")
		auth.Use(jwt.JWTAuthMiddleware())
		{
			// 监管类型
			auth.POST("/standard-phrase/supervision-type", api.AddSupervisionType)
			auth.PUT("/standard-phrase/supervision-type", api.UpdateSupervisionType)
			auth.DELETE("/standard-phrase/supervision-type/:ids", api.DeleteSupervisionType)

			// 规范类别
			auth.POST("/standard-phrase/category", api.AddCategory)
			auth.PUT("/standard-phrase/category", api.UpdateCategory)
			auth.DELETE("/standard-phrase/category/:ids", api.DeleteCategory)

			// 规范条目
			auth.POST("/standard-phrase/item", api.AddItem)
			auth.PUT("/standard-phrase/item", api.UpdateItem)
			auth.DELETE("/standard-phrase/item/:ids", api.DeleteItem)

			// 规范内容
			auth.POST("/standard-phrase/content", api.AddContent)
			auth.PUT("/standard-phrase/content", api.UpdateContent)
			auth.DELETE("/standard-phrase/content/:ids", api.DeleteContent)
		}
	}
}

File: frontend/health-enforcement-ui/src/api/system/standardPhrase.js
import request from '@/utils/request'
import { parseStrEmpty } from "@/utils/ruoyi";

// ============ 监管类型 API ============

// 查询监管类型列表
export function listSupervisionType(query) {
  return request({
    url: '/system/standard-phrase/supervision-type/list',
    method: 'get',
    params: query
  })
}

// 查询监管类型详细
export function getSupervisionType(id) {
  return request({
    url: '/system/standard-phrase/supervision-type/' + parseStrEmpty(id),
    method: 'get'
  })
}

// 新增监管类型
export function addSupervisionType(data) {
  return request({
    url: '/system/standard-phrase/supervision-type',
    method: 'post',
    data: data
  })
}

// 修改监管类型
export function updateSupervisionType(data) {
  return request({
    url: '/system/standard-phrase/supervision-type',
    method: 'put',
    data: data
  })
}

// 删除监管类型
export function deleteSupervisionType(ids) {
  return request({
    url: '/system/standard-phrase/supervision-type/' + ids,
    method: 'delete'
  })
}

// ============ 规范类别 API ============

// 查询规范类别列表
export function listCategory(query) {
  return request({
    url: '/system/standard-phrase/category/list',
    method: 'get',
    params: query
  })
}

// 查询规范类别详细
export function getCategory(id) {
  return request({
    url: '/system/standard-phrase/category/' + parseStrEmpty(id),
    method: 'get'
  })
}

// 新增规范类别
export function addCategory(data) {
  return request({
    url: '/system/standard-phrase/category',
    method: 'post',
    data: data
  })
}

// 修改规范类别
export function updateCategory(data) {
  return request({
    url: '/system/standard-phrase/category',
    method: 'put',
    data: data
  })
}

// 删除规范类别
export function deleteCategory(ids) {
  return request({
    url: '/system/standard-phrase/category/' + ids,
    method: 'delete'
  })
}

// ============ 规范条目 API ============

// 查询规范条目列表
export function listItem(query) {
  return request({
    url: '/system/standard-phrase/item/list',
    method: 'get',
    params: query
  })
}

// 查询规范条目详细
export function getItem(id) {
  return request({
    url: '/system/standard-phrase/item/' + parseStrEmpty(id),
    method: 'get'
  })
}

// 新增规范条目
export function addItem(data) {
  return request({
    url: '/system/standard-phrase/item',
    method: 'post',
    data: data
  })
}

// 修改规范条目
export function updateItem(data) {
  return request({
    url: '/system/standard-phrase/item',
    method: 'put',
    data: data
  })
}

// 删除规范条目
export function deleteItem(ids) {
  return request({
    url: '/system/standard-phrase/item/' + ids,
    method: 'delete'
  })
}

// ============ 规范内容 API ============

// 查询规范内容列表
export function listContent(query) {
  return request({
    url: '/system/standard-phrase/content/list',
    method: 'get',
    params: query
  })
}

// 查询规范内容详细
export function getContent(id) {
  return request({
    url: '/system/standard-phrase/content/' + parseStrEmpty(id),
    method: 'get'
  })
}

// 新增规范内容
export function addContent(data) {
  return request({
    url: '/system/standard-phrase/content',
    method: 'post',
    data: data
  })
}

// 修改规范内容
export function updateContent(data) {
  return request({
    url: '/system/standard-phrase/content',
    method: 'put',
    data: data
  })
}

// 删除规范内容
export function deleteContent(ids) {
  return request({
    url: '/system/standard-phrase/content/' + ids,
    method: 'delete'
  })
}

// ============ 搜索 API ============

// 搜索规范用语
export function searchStandardPhrase(query) {
  return request({
    url: '/system/standard-phrase/search',
    method: 'get',
    params: query
  })
}

// 获取完整树形结构
export function getFullTree(query) {
  return request({
    url: '/system/standard-phrase/tree',
    method: 'get',
    params: query
  })
}
